query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -136,10 +136,16 @@ private String convertFilterToString(
       String delim = "";
       builder.append("( ");
       for (Filter childFilter : filter.getChildFilterList()) {
-        builder.append(delim);
-        builder.append(convertFilterToString(childFilter, paramsBuilder, executionContext));
-        builder.append(" ");
-        delim = operator + " ";
+        // child filter should either have further child filters or should have LHS and RHS
+        // only then child filter is converted to string
+        // this check avoids converting empty filters to string
+        if (!childFilter.getChildFilterList().isEmpty()
```

Comment:
These checks should go as part of query transformation that happens before actual query execution. See `AbstractQueryTransformation.transform()`. There is already transformFilter() that skips certain defaults.

Reply:
+1

Reply:
done